### PR TITLE
Migrate from k8s.gcr.io to registry.k8s.io

### PR DESCRIPTION
**Pull request by harshitasao**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR migrate from k8s.gcr.io to registry.k8s.io

**Which issue(s) this PR fixes**:
Fixes  https://github.com/kubernetes/k8s.io/issues/4780 

**Special notes for your reviewer**:

This fix is a part of the Umbrella issue https://github.com/kubernetes/k8s.io/issues/4780

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/lgtm

**Comment by Poor12** [comment]
Great thanks! I think we do not need to change in previous versions of the document, such as 1.3 and 1.4.

File: docs/troubleshooting/trouble-shooting.md [hunk 1]
```diff
@@ -4,14 +4,14 @@ title: Troubleshooting
 
 ## I can't access some resources when installing Karmada
 
-- Pull images from Google Container Registry (k8s.gcr.io).
+- Pull images from Google Container Registry (registry.k8s.io).
```
Comment (RainbowMango) [inline docs/troubleshooting/trouble-shooting.md:7]: `Google Container Registry`? I think the `registry.k8s.io` now acts the agent, right?

File: i18n/zh/docusaurus-plugin-content-docs/version-v1.4.0/troubleshooting/trouble-shooting.md [hunk 2]
```diff
@@ -4,14 +4,14 @@ title: 故障定位
 
 ## 我在安装 Karmada 时无法访问一些资源
 
-- 从谷歌容器仓库（k8s.gcr.io）拉取镜像。
+- 从谷歌容器仓库（registry.k8s.io）拉取镜像。
```
Comment (RainbowMango) [inline i18n/zh/docusaurus-plugin-content-docs/version-v1.4.0/troubleshooting/trouble-shooting.md:7]: ditto `谷歌容器仓库`

File: versioned_docs/version-v1.3.0/troubleshooting/troubleshooting.md [hunk 3]
```diff
@@ -4,14 +4,14 @@ title: Troubleshooting
 
 ## I can't access some resources when installing Karmada
 
-- Pull images from Google Container Registry (k8s.gcr.io).
+- Pull images from Google Container Registry (registry.k8s.io).
```
Comment (RainbowMango) [inline versioned_docs/version-v1.3.0/troubleshooting/troubleshooting.md:7]: ditto

File: versioned_docs/version-v1.4.0/troubleshooting/trouble-shooting.md [hunk 4]
```diff
@@ -4,14 +4,14 @@ title: Troubleshooting
 
 ## I can't access some resources when installing Karmada
 
-- Pull images from Google Container Registry (k8s.gcr.io).
+- Pull images from Google Container Registry (registry.k8s.io).
```
Comment (RainbowMango) [inline versioned_docs/version-v1.4.0/troubleshooting/trouble-shooting.md:7]: ditto

**Review by RainbowMango — APPROVED** [review]
/approve

Please @Poor12 help to fix the comments after this.

**Pull request merged**